src/test/java/uk/gov/companieshouse/controller/AppealControllerTest.java

```diff
@@ -0,0 +1,200 @@
+            .thenReturn(TEST_RESOURCE_ID);
+    }
+
+    @Test
+    public void whenValidInput_return201() throws Exception {
+
+        String appealAsJson = asJsonString(validAppeal());
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .headers(createHttpHeaders())
+            .content(appealAsJson))
+            .andExpect(status().isCreated())
+            .andExpect(header().string(HttpHeaders.LOCATION,"http://localhost/companies/12345678/appeals/" + TEST_RESOURCE_ID));
+    }
+
+    @Test
+    public void whenNullEricIdentityHeader_return400() throws Exception {
+
+        String appealAsJson = asJsonString(validAppeal());
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(appealAsJson))
+            .andExpect(status().isBadRequest())
+            .andExpect(status().reason("Missing request header 'ERIC-identity' for method parameter of type String"));
+    }
+
+    @Test
+    public void whenBlankEricIdentityHeader_return401() throws Exception {
+
+        String appealAsJson = asJsonString(validAppeal());
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .header(IDENTITY_HEADER, "")
+            .content(appealAsJson))
+            .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void whenNullRequestBody_return400() throws Exception {
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .headers(createHttpHeaders()))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void whenInvalidInput_return422() throws Exception {
+
+        Appeal appeal = validAppeal();
+        appeal.setPenaltyIdentifier(null);
+
+        String appealAsJson = asJsonString(appeal);
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .headers(createHttpHeaders())
+            .content(appealAsJson))
+            .andExpect(status().isUnprocessableEntity())
+            .andExpect(content().json("{'penaltyIdentifier':'penaltyIdentifier must not be null'}"));
+    }
+
+    @Test
+    public void whenExceptionFromService_return500() throws Exception {
+
+        when(appealService.saveAppeal(any(String.class), any(Appeal.class), any(String.class)))
+            .thenThrow(Exception.class);
+
+        String appealAsJson = asJsonString(validAppeal());
+
+        mockMvc.perform(post(APPEALS_URI, TEST_COMPANY_ID)
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .headers(createHttpHeaders())
+            .content(appealAsJson))
+            .andExpect(status().isInternalServerError());
+    }
+
+    @Test
+    public void whenAppealExists_return200() throws Exception {
```

Comment:
I'd suggest splitting tests into separate file so that each HTTP endpoint has tests in it's own file.

Reply:
done